### ISSUE TITLE
Usage listed as compliant in documentation for UnnecessaryLet not working as expected.

### ISSUE THREAD

**Issue by jck2** [detekt/detekt#5286]
I have a usage of let that looks to me like it takes the a form similar to `a?.let { print(it) }`, which is listed as compliant in the docs but it is flagged as an error so I'm wondering if I'm missing a subtle part of the rule here.

I have a data class with a few nullable properties, for the sake of example say it's something like:
```
data class foo(val title: String?)
``` 
I also have an interface that defines a function setTitle that looks something like: 
```
interface bar { 
  fun setTitle(title:String) 
}
```

The code block in my code essentially looks like the following:
```
val a : foo = ...
with(a) {
  title?.let { bar.setTitle(it) }
}
```

Is this a valid use case for let or am I not understanding the docs correctly? Also I wasn't sure if this was an improper usage of with, but if it is, I'd expect a different error to be reported by detekt

**Comment by jck2** [comment]
Seems like something weird was happening with the detekt output. There was another detekt issue in a different file and it flagged the lets in my file, but fixing the detekt error in the other file resolved this problem. Will close
